File: admin front/src/types/category.ts
export interface AdminMarketSector {
    id: number;
    name: string;
    slug: string;
    description?: string;
    iconKey?: string;
    imageUrl?: string;
    sortOrder: number;
    isActive: boolean;
    categoriesCount: number;
    createdAt: string;
}

export interface AdminCategory {
    id: number;
    marketSectorId: number;
    marketSectorName?: string;
    marketSectorSlug?: string;
    name: string;
    slug: string;
    description?: string;
    imageUrl?: string;
    sortOrder: number;
    isActive: boolean;
    createdAt: string;
}

export interface CreateAdminCategoryDto {
    name: string;
    marketSectorId: number;
    description?: string;
    image?: File;
    sortOrder?: number;
    isActive?: boolean;
}

export interface UpdateAdminCategoryDto {
    name: string;
    marketSectorId: number;
    description?: string;
    image?: File;
    sortOrder?: number;
    isActive?: boolean;
}

export interface ReorderAdminCategoryDto {
    id: number;
    sortOrder: number;
}

export interface CreateAdminMarketSectorDto {
    name: string;
    description?: string;
    iconKey?: string;
    image?: File;
    sortOrder?: number;
    isActive?: boolean;
}

export type UpdateAdminMarketSectorDto = CreateAdminMarketSectorDto;

File: CityOrders.Api/Domain/Entities/Category.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CityOrders.Api.Domain.Entities
{
    public class Category
    {
        [Key]
        public int Id { get; set; }

        public int MarketSectorId { get; set; } = 1;
        public MarketSector MarketSector { get; set; } = null!;

        [Required]
        [MaxLength(100)]
        public string Name { get; set; } = string.Empty;

        [Required]
        [MaxLength(120)]
        public string Slug { get; set; } = string.Empty;

        [MaxLength(500)]
        public string? Description { get; set; }

        [MaxLength(80)]
        public string? IconKey { get; set; }

        [MaxLength(500)]
        public string? ImageUrl { get; set; }

        public int SortOrder { get; set; } = 0;

        public bool IsActive { get; set; } = true;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        public ICollection<Brand> Brands { get; set; } = new List<Brand>();
    }
}
